data/human/paradise world jobs.txt

```diff
@@ -521,3 +521,584 @@ mission "Paradise Job: Wilderness Retreat"
+	job
+	repeat
+	cargo Medical 5 25
+	clearance "When you explain to the spaceport controller that you are carrying a vital supply of <commodity> to help treat the wounded, they give you the clearance to land."
+	to offer
+		has "event: war begins"
+		not "event: initial deployment 2"
+		random < 15
+	source
+		near Pollux 2 10
+	destination Martini
+	on visit
+		dialog phrase "generic cargo on visit"
+	on complete
+		payment 25000 100
+		dialog phrase "generic cargo delivery payment"
+
+
+mission "Martini Bombings [Reconstruction]"
+	name "Construction supplies to Martini"
+	description "Due to the recent nuclear bombings, bring <cargo> to <planet> to help the reconstruction efforts. Payment is <payment>."
+	job
+	repeat
+	cargo Construction 5 25
+	clearance "When you explain to the spaceport controller that you are carrying a vital supply of <commodity> for the reconstruction efforts, they give you the clearance to land."
+	to offer
+		has "event: war begins"
+		not "event: initial deployment 2"
+		random < 30
+	source
+		near Pollux 2 10
+	destination Martini
+	on visit
+		dialog phrase "generic cargo on visit"
+	on complete
+		payment 30000 50
+		dialog phrase "generic cargo delivery payment"
+
+
+mission "Martini Bombings [Food]"
+	name "Food packages to Martini"
+	description "Due to the recent nuclear bombings, bring <cargo> to <planet> to help keep up supply. Payment is <payment>."
+	job
+	repeat
+	cargo "food packages" 5 25
+	clearance "When you explain to the spaceport controller that you are carrying a vital supply of <commodity> for the refugee camps, they give you the clearance to land."
+	to offer
+		has "event: war begins"
+		not "event: initial deployment 2"
+		random < 30
+	source
+		near Pollux 2 10
+	destination Martini
+	on visit
+		dialog phrase "generic cargo on visit"
+	on complete
+		payment 30000 50
+		dialog phrase "generic cargo delivery payment"
+
+
+mission "Martini Bombings [Water Tablets]"
+	name "Water purification tablets to Martini"
+	description "Due to the recent nuclear bombings, bring <cargo> to <planet> to help keep up supply. Payment is <payment>."
+	job
+	repeat
+	cargo "water purification tablets" 5 25
+	clearance "When you explain to the spaceport controller that you are carrying a vital supply of water purification tablets for the refugee camps, they give you the clearance to land."
+	to offer
+		has "event: war begins"
+		not "event: initial deployment 2"
+		random < 30
+	source
+		near Pollux 2 10
+	destination Martini
+	on visit
+		dialog phrase "generic cargo on visit"
+	on complete
+		payment 30000 50
+		dialog phrase "generic cargo delivery payment"
+
+
+mission "Martini Bombings [Relief People 1]"
+	name "Relief workers to Martini"
+	description "Due to the recent nuclear bombings on Martini, bring <bunks> <martini relief occupations> to <planet> to release demand from overloaded hospitals. Payment is <payment>."
+	job
+	repeat
+	passengers 5 15
+	substitutions
```

Comment:
Technical question here for the mission patrol. Is this replacement guaranteed to be the same for the description and the clearance? Also, do those numbers check out? I don't think that actually gives 25% chance to each option.

Reply:
This substitution arrangement was suggested to me by RisingLeaf on discord. I don't see the problem here though, but I don't know how random works exactly.

I still need to test this, I was going to tonight but I wasn't able to. I will check tomorrow morning.

Reply:
I'm not exactly sure how they work, but I seem to recall that all cases are evaluated. So, first we set the replacement to the first value. Then, with a 75% chance, to the second value. After that, regardless of the previous result, we update it with a 50% chance. That means there is only a `1-0.75-(0.5-0.5*0.75)-(0.25-0.25*0.5-0.25*0.75+0.25*0.5*0.75)=0.09375`, or 9.3% chance of choosing the first option. That assumes I remember the lessons on probability theory correctly.

Reply:
The substitution considers the bottommost one first, not the topmost.

Reply:
That doesn't seem right either. The progression should be 1/4->1/3->1/2->1, insteaed of 1/4->1/2->3/4->1 (assuming it stops at the first match (otherwise it would always be the default case)).

Reply:
I'll look up the conversation I had with RisingLeaf about this.

Reply:
Hmm, it doesn't say much. I will try it out in-game anyway and see.

Reply:
It does roughly seem to work, but yeah, tibetiroka has the right idea for how to evenly spread it out. If the first 25% roll doesn't go through, you've got the remaining 75% to work with. A second 25% chance in that 75% is 1/3 of that. If _that_ doesn't roll, then we're just left with 50% of the original chance, where a 25% chance splits that in half.